### PR TITLE
[ci][v2/2] split lint into sub buildkite steps

### PR DESCRIPTION
**Pull request by can-anyscale**
Split lint into sub buildkite steps, so that we can catch more issues, as well as make the errors more readable

Test strategies:
- CI tests

### CODE REVIEW

**Comment by can-anyscale** [comment]
beautifully green

Screenshot 2023-07-18 at 3 29 20 PM

File: ci/v2/lint.sh [hunk 1]
```diff
@@ -16,28 +16,19 @@ lint_readme() {
   fi
 }
 
-lint_scripts() {
+scripts() {
```
Comment (aslonnie) [inline ci/v2/lint.sh:19]: what type of scripts? maybe rename to `shell_scripts`?

**Review by aslonnie — COMMENTED** [review]
<div><br/>via <a href="https://giphy.com/gifs/kill-me-now-p8GJOXwSNzQPu">GIPHY</a></div>

**Review by aslonnie — APPROVED** [review]

File: ci/v2/lint.sh (hunk 1 above)
Comment (can-anyscale) [reply ci/v2/lint.sh]: changed to code_format, I think that's what it is (shell, python, java, cpp)

**Pull request merged**